subprojects/base-services/src/main/java/org/gradle/internal/UncheckedException.java

```diff
@@ -44,6 +44,9 @@ public static RuntimeException throwAsUncheckedException(Throwable t) {
      * Note: always throws the failure in some form. The return value is to keep the compiler happy.
      */
     public static RuntimeException throwAsUncheckedException(Throwable t, boolean preserveMessage) {
+        if (t instanceof InterruptedException) {
+            Thread.currentThread().interrupt();
+        }
```

Comment:
Ooh, this is sneaky... I think this behavior should at least be mentioned in the Javadoc.

Reply:
Sorry, random fly on the wall here, saw an interesting PR and decided to browse.

TBH, anything that _doesn't_ do this is kind of a red flag to me.  Unless you are explicitly interrupting yourself and handling thread lifecycle (i.e. expecting interruption because you are signalling your code to stop), it's best to reset the flag whenever rethrowing interruptions.  If you don't know who calls you, always reset.